src/common/contracts/staking-router/staking-router-fetch.service.ts

```diff
@@ -1,44 +1,68 @@
-import { StakingRouter, STAKING_ROUTER_CONTRACT_TOKEN } from '@lido-nestjs/contracts';
 import { Injectable, Inject, LoggerService } from '@nestjs/common';
 import { StakingModule } from './staking-module';
 import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
 import { IStakingModuleService } from 'common/contracts/i-staking-module';
 import { STAKING_MODULE_TYPE } from 'staking-router-modules';
+import { LidoLocatorService } from 'common/contracts/lido-locator';
+import { StakingRouter__factory } from 'generated';
+import { ExecutionProvider } from 'common/execution-provider';
 
 @Injectable()
 export class StakingRouterFetchService {
   constructor(
-    @Inject(STAKING_ROUTER_CONTRACT_TOKEN) protected readonly contract: StakingRouter,
     @Inject(LOGGER_PROVIDER) protected readonly loggerService: LoggerService,
     protected readonly iStakingModule: IStakingModuleService,
+    protected readonly lidoLocatorService: LidoLocatorService,
+    protected readonly provider: ExecutionProvider,
   ) {}
 
+  private getContract(contractAddress: string) {
+    return StakingRouter__factory.connect(contractAddress, this.provider);
+  }
+
   /**
    *
    * @returns Staking Router modules list
    */
   public async getStakingModules(blockTag: number | string): Promise<StakingModule[]> {
-    const modules = await this.contract.getStakingModules({ blockTag });
+    const stakingRouterAddress = await this.lidoLocatorService.getStakingRouter(blockTag);
+
+    this.loggerService.debug?.('Lido Locator', stakingRouterAddress);
```

Comment:
it would be better to change log message

Reply:
fixed